test/object-store/util/sync/redirect_server.hpp

```diff
@@ -60,9 +84,59 @@ class RedirectingHttpServer {
         m_server_thread.join();
     }
 
+    void set_event_hook(std::function<void(Event, std::optional<std::string>)> hook)
+    {
+        m_hook = hook;
+    }
+
+    // If true, http (app services) requests will first hit the redirect server and
+    // receive a redirect response which will contain the location to the actual
+    // server.
+    // NOTE: some http transport redirect implementations may strip the authorization
+    // header from the request after it is redirected and the user will be logged out
+    // from the client app as a result.
+    void force_http_redirect(bool remote = true)
```

Comment:
It seems like this is always called with an argument, do you need a default argument value here?

Reply:
Not necessary (thought was that calling `force_http_redirect()` would do just that - removed default